src/types/ant.ts

```diff
@@ -202,109 +203,49 @@ export interface AoANTRead {
-    },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  removeController(
-    {
-      controller,
-    }: {
-      controller: WalletAddress;
-    },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setRecord(
-    {
-      undername,
-      transactionId,
-      ttlSeconds,
-    }: {
-      undername: string;
-      transactionId: string;
-      ttlSeconds: number;
-    },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  removeRecord(
-    { undername }: { undername: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setBaseNameRecord({
-    transactionId,
-    ttlSeconds,
-  }: {
-    transactionId: string;
-    ttlSeconds: number;
-  }): Promise<AoMessageResult>;
-  setUndernameRecord({
-    undername,
-    transactionId,
-    ttlSeconds,
-  }: {
-    undername: string;
-    transactionId: string;
-    ttlSeconds: number;
-  }): Promise<AoMessageResult>;
-  removeUndernameRecord({
-    undername,
-  }: {
-    undername: string;
-  }): Promise<AoMessageResult>;
-  setTicker(
-    { ticker }: { ticker: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setDescription(
-    { description }: { description: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setKeywords(
-    { keywords }: { keywords: string[] },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setName(
-    { name }: { name: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  setLogo(
-    { txId }: { txId: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  releaseName(
-    { name, arioProcessId }: { name: string; arioProcessId: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  reassignName(
-    {
-      name,
-      arioProcessId,
-      antProcessId,
-    }: { name: string; arioProcessId: string; antProcessId: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  approvePrimaryNameRequest(
-    {
-      name,
-      address,
-      arioProcessId,
-    }: { name: string; address: WalletAddress; arioProcessId: string },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
-  removePrimaryNames(
-    {
-      names,
-      arioProcessId,
-      notifyOwners,
-    }: { names: string[]; arioProcessId: string; notifyOwners?: boolean },
-    options?: WriteOptions,
-  ): Promise<AoMessageResult>;
+  transfer: WriteAction<{ target: WalletAddress }>;
+  addController: WriteAction<{ controller: WalletAddress }>;
+  removeController: WriteAction<{ controller: WalletAddress }>;
+  /** @deprecated Use setUndernameRecord instead for undernames, and setBaseNameRecord instead for the top level name (e.g. "@") */
+  setRecord: WriteAction<AoANTSetUndernameRecordParams>;
+  removeRecord: WriteAction<{ undername: string }>;
+  setBaseNameRecord: WriteAction<AoANTSetBaseNameRecordParams>;
+  setUndernameRecord: WriteAction<AoANTSetUndernameRecordParams>;
```

Comment:
I found that these new methods did not have `WriteOptions` copy pasted into the parameters. to prevent this in the future I created a new `WriteAction` utility type that will apply the options and the common result. fwiw its only been applied on the ANT Type to consolidate the diff

Reply:
also just noticed that it wasn't on the implementation, added here : df830b6